### ISSUE TITLE
Allow inverse-normalize data using tf.normalize() function

### ISSUE THREAD

**Issue by neel04** [tensorflow/tensorflow#45788]
**System information**
- TensorFlow version (you are using): 2.3.0
- Are you willing to contribute it (Yes/No): No

I am working a Regression(seq-2-seq) problem using a custom model, and while building my Data Pipeline I used `tf.keras.utilities.normalize` which automatically normalizes a numpy array in a feauture range not specified by the user, it works very well in normalizing the data.

However, after training a model on the normalized data (Where the `labels` were normalized using this utility), I am getting predictions as "normalized" labels. Apparently, when I run the `.evaluate()` function to test out my model, I am getting an `RMSE: 0.00089` which indicates that the model is performing well and is **de-normalizing the labels during evaluation**. However when I predict on 1 sample, the predictions are not normalized.

So, in short, can a feature be introduced that allows the function to de-normalize a numpy array manually (Like a Boolean argument) perhaps inheriting the required metadata from the `model` object? I think it would be very helpful for everyone using this function and would aid in quick&clean normalization.

**Comment by shrikumaran** [comment]
Can I help in coding this feature ?

**Comment by neel04** [comment]
@rmothukuru Any workaround for the time being?

**Comment by rmothukuru** [comment]
@neel04,
In the below statement:

> However when I predict on 1 sample, the predictions are not normalized. 

do you mean the Predictions are **not Denormalized**?

Also, In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by neel04** [comment]
@rmothukuru There is no issue - it is just that if I normalize data with that function, there is simply no way to recover the original data (before normalization)

**Comment by fchollet** [comment]
To denormalize an array, you need to know the mean and variance that were used to normalize it in the first place. This is not the case here.

Also note that should not use `evaluate()` to compute RMSE because `evaluate()` computes metrics sample-wise then averages the per-sample results. However, the square root of a sum is not the sum of the square roots.
